web/src/scheduled_messages.js

```diff
@@ -0,0 +1,158 @@
+        narrow.activate([{operator: "pm-with", operand: compose_args.private_message_recipient}], {
+            trigger: "edit scheduled message",
+        });
+    }
+
+    overlays.close_overlay("scheduled");
+    compose_fade.clear_compose();
+    compose.clear_preview_area();
+    compose_actions.start(compose_args.type, compose_args);
+    compose_ui.autosize_textarea($("#compose-textarea"));
+    $("#compose-textarea").attr("data-scheduled-message-id", scheduled_msg_id);
+}
+
+function format(scheduled_messages) {
+    for (const msg of scheduled_messages) {
+        if (msg.type === "stream") {
+            msg.is_stream = true;
+            msg.stream_id = msg.to[0];
+            msg.stream_name = stream_data.maybe_get_stream_name(msg.stream_id);
+            const color = stream_data.get_color(msg.stream_name);
+            msg.recipient_bar_color = stream_color.get_recipient_bar_color(color);
+            msg.stream_privacy_icon_color = stream_color.get_stream_privacy_icon_color(color);
+        } else {
+            msg.is_stream = false;
+            msg.recipients = people.get_recipients(msg.to.join(","));
+        }
+        const time = new Date(msg.deliver_at);
+        msg.time_stamp = timerender.time_left_to_achieve_date(time);
+        msg.full_date_time = timerender.get_full_datetime(time);
+    }
+}
+
+export function delete_scheduled_message(scheduled_msg_id) {
+    channel.del({
+        url: "/json/scheduled_messages/" + scheduled_msg_id,
+        success() {
+            if (overlays.scheduled_messages_open()) {
+                $(
+                    `#scheduled_messages_overlay .scheduled-message-row[data-message-id=${scheduled_msg_id}]`,
+                ).remove();
+            }
+        },
+    });
+}
+
+export function delete_scheduled_message_if_sent_directly() {
+    // Delete old scheduled message if it was sent.
+    if ($("#compose-textarea").attr("data-scheduled-message-id")) {
+        delete_scheduled_message($("#compose-textarea").attr("data-scheduled-message-id"));
+        $("#compose-textarea").removeAttr("data-scheduled-message-id");
+    }
+}
+
+export function launch() {
+    $("#scheduled_messages").empty();
+    $("#scheduled_messages").append(render_scheduled_messages_overlay());
+    overlays.open_overlay({
+        name: "scheduled",
+        $overlay: $("#scheduled_messages_overlay"),
+        on_close() {
+            browser_history.exit_overlay();
+        },
+    });
+    loading.make_indicator($("#scheduled_messages_overlay .loading-indicator"), {
+        abs_positioned: true,
+    });
+
+    channel.get({
+        url: "/json/scheduled_messages",
+        success(data) {
+            hide_loading_indicator();
+            if (data.scheduled_messages.length === 0) {
+                $(".no-overlay-messages").show();
+            } else {
+                data.scheduled_messages.sort(
+                    (first, second) => first.deliver_at - second.deliver_at,
+                );
```

Comment:
Given that you're doing this sorting here, I wonder if we should adjust the server part to return them in sorted order by that field; it seems like it's what most clients would want/expect, and might not be hard to do.

Reply:
Yeah, just used `order_by` for this.

Reply:
Looks good, and also I updated the API docs to mention this while I was fiddling with this.